### PR TITLE
Added trait scores support.

### PR DESCRIPTION
**Pull request by agorina**

### CODE REVIEW

File: exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/model/ExamItem.java [hunk 1]
```diff
@@ -1,26 +1,28 @@
...
+    private final static int OrganizationAndPurpose = 2;
+    private final static int Conventions = 3;
+
+    private String naturalId;
+    private Double score;
+    private String scoreStatus;
+    private String response;
+    private Double maxScore;
+    private Long position;
+    private Map<Integer, ExamItemSubScore> subScoresBytraitId;
```
Comment (malaffoon) [inline exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/model/ExamItem.java]: A map seems like overkill here. At most there are 3 of these right? A list would be more efficient; yeah, you'd want to write a little helper to replace the containsKey() call but that is easy.

File: exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultExamItemProcessor.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    private final Map<String, Integer> itemTraitIds;
+
+    /**
+     * Constructor
+     *
+     * @param repository the {@link ItemTraitRepository}
+     */
+    @Autowired
+    public DefaultExamItemProcessor(final ItemTraitRepository repository) {
```
Comment (malaffoon) [inline exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultExamItemProcessor.java]: in general should these autowired constructors be package-private?

File: exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/model/ExamItem.java (hunk 1 above)
Comment (agorina) [inline exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/model/ExamItem.java]: Really? I would think that the map implementation should be smart enough. You made me curious. I will check it out.

File: exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultExamItemProcessor.java (hunk 2 above)
Comment (agorina) [inline exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultExamItemProcessor.java]: You are right. Will change. Thank you.

File: exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/model/ExamItem.java (hunk 1 above)
Comment (malaffoon) [inline exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/model/ExamItem.java]: Smart? Sure, maps are smart. But they are big and slow compared to a list, when dealing with less than a couple dozen things. They also complicate marshaling, etc. although that might not be an issue here.

**Pull request merged**